src/Farmer/Deploy.fs

```diff
@@ -186,8 +191,8 @@ let setSubscription (subscriptionId:Guid) =
     Az.setSubscription (subscriptionId.ToString())
 
 /// Validates that the parameters supplied meet the deployment requirements.
-let validateParameters suppliedParameters deployment =
-    let expected = deployment.Template.Parameters |> List.map(fun (SecureParameter p) -> p) |> Set
+let validateParameters suppliedParameters (deployment:#IDeploymentSource) =
```

Comment:
Is there a reason why we're using flexible types here (i.e. `#IDeploymentSource` instead of just `IDeploymentSource`)? I'm not sure it's required.

Reply:
This applies for all cases this file.